src/main/java/org/jenkinsci/plugins/xunit/pipeline/XUnitResultsStep.java

```diff
@@ -0,0 +1,157 @@
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.jenkinsci.plugins.xunit.Messages;
+import org.jenkinsci.plugins.xunit.XUnitDefaultValues;
+import org.jenkinsci.plugins.xunit.XUnitUtil;
+import org.jenkinsci.plugins.xunit.threshold.FailedThreshold;
+import org.jenkinsci.plugins.xunit.threshold.SkippedThreshold;
+import org.jenkinsci.plugins.xunit.threshold.XUnitThreshold;
+import org.jenkinsci.plugins.xunit.threshold.XUnitThresholdDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+public class XUnitResultsStep extends Step {
+    private List<TestType> tools;
+    private List<XUnitThreshold> thresholds;
+    private int thresholdMode = 1;
+    private long testTimeMargin = XUnitDefaultValues.TEST_REPORT_TIME_MARGING;
+    private Collection<TestDataPublisher> testDataPublishers;
+
+    @DataBoundConstructor
+    public XUnitResultsStep(@CheckForNull List<TestType> tools) {
+        if (tools != null) {
+            this.tools = tools;
+        } else {
+            this.tools = new ArrayList<>();
+        }
+    }
+
+    @Nonnull
+    public List<TestType> getTools() {
+        return tools;
+    }
+
+    @Nonnull
+    public List<XUnitThreshold> getThresholds() {
+        return thresholds != null ? thresholds : Collections.emptyList();
+    }
+
+    @DataBoundSetter
+    public void setThresholds(@Nonnull List<XUnitThreshold> thresholds) {
+        this.thresholds = thresholds;
+    }
+
+    @DataBoundSetter
+    public void setThresholdMode(int thresholdMode) {
+        this.thresholdMode = thresholdMode;
+    }
+
+    public int getThresholdMode() {
+        return thresholdMode;
+    }
+
+    @DataBoundSetter
+    public void setTestTimeMargin(String testTimeMargin) {
+        this.testTimeMargin = XUnitUtil.parsePositiveLong(testTimeMargin, XUnitDefaultValues.TEST_REPORT_TIME_MARGING);
+    }
+
+    public String getTestTimeMargin() {
+        return String.valueOf(testTimeMargin);
+    }
+
+    public long getTestTimeMarginAsLong() {
+        return testTimeMargin;
+    }
+
+    @Nonnull
+    public Collection<TestDataPublisher> getTestDataPublishers() {
+        return testDataPublishers != null ? testDataPublishers : Collections.emptyList();
+    }
+
+    @DataBoundSetter
+    public void setTestDataPublishers(@Nonnull Collection<TestDataPublisher> testDataPublishers) {
+        this.testDataPublishers = testDataPublishers;
+    }
+
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        return new XUnitResultsStepExecution(this, context);
+    }
+
+    @Extension
+    public static class DescriptorImpl extends StepDescriptor {
```

Comment:
This descriptor lacks of 
```
public DescriptorExtensionList<TestDataPublisher, Descriptor<TestDataPublisher>> getListTestDataPublisherDescriptors() {
    return hudson.tasks.junit.TestDataPublisher.all();
}
```
Without this method the snippet does not show test data publishers

Reply:
Good catch - I forgot to copy that over.